### PR TITLE
Enhancement/7061-hide-kmw-tiles-before-setup

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #7061 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/datastore/user/key-metrics.js [hunk 1]
```diff
@@ -257,6 +257,26 @@ const baseSelectors = {
...
 
+	/**
+	 * Gets whether an individual key metric identified by its slug is
+	 * active or not.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean|undefined} True if the key metric widget tile is active, false if it is not, or undefined if the key metrics settings are not loaded.
+	 */
+	isKeyMetricActive: createRegistrySelector(
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/datastore/user/key-metrics.js:268]: Note, every module asset that uses this should be modified to depend on the core/user store's asset now. I can see at least GA4 currently does not
https://github.com/google/site-kit-wp/blob/37352fca946b60a142a4aea0cc19d0e056fc9285/includes/Modules/Analytics_4.php#L933-L945

**Review by nfmohit — APPROVED** [review]
Excellent work here, LGTM, thank you @jimmymadon!

Just a note that there was a merge conflict with the `TopConvertingTrafficSourceWidget` widget updated as a part of another issue.

I have resolved the merged conflict, but also added a few follow-up commits that would've ideally been a part of the merge commit itself.

File: assets/js/components/DashboardMainApp.js [hunk 2]
```diff
@@ -157,12 +162,12 @@ export default function DashboardMainApp() {
...
 				render when `userInputEnabled` is false.
 
-				This check can be removed once the User Input feature is fully launched
+				The userInputEnabled check can be removed once the User Input feature is fully launched
 				and we remove this feature flag.
 
 				See: https://github.com/google/site-kit-wp/pull/6630#discussion_r1127229162
 			*/ }
-			{ userInputEnabled && (
+			{ userInputEnabled && isKeyMetricsWidgetHidden !== true && (
```
Comment (techanvil) [inline assets/js/components/DashboardMainApp.js:170]: Hey @jimmymadon, nice work on this PR.

Looking at this condition, I'm wondering if it should be tweaked so the `WidgetContextRenderer` is not rendered when `isKeyMetricsWidgetHidden` is `undefined`, i.e. the state is still resolving. To be honest in my testing this seems to be working fine in practice, so I don't want to throw a red herring in here, but it feels like the safer thing to do would be avoid rendering these widgets at the top level until the `isKeyMetricsWidgetHidden` state is resolved, so we don't initially render them and then remove once the `isKeyMetricsWidgetHidden` is resolved to `true`.

In other words we could update as follows:

```suggestion
			{ userInputEnabled && isKeyMetricsWidgetHidden === false && (
```

Have you already considered this, and what do you think?

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @jimmymadon, nice work here. I've left a couple of related questions for you, please take a look and see what you think.

File: assets/js/components/DashboardMainApp.js (hunk 2 above)
Comment (techanvil) [reply assets/js/components/DashboardMainApp.js]: Hey @jimmymadon, thanks for the background here and for making the change. However, and I must apologise because I could have saved us some time here if I'd only spotted this the first time around - this change does actually introduce a bug, whereby the Key Metrics navigation chip is no longer correctly selected on page load. Here's a screen capture to demonstrate:

https://github.com/google/site-kit-wp/assets/18395600/5bb271b5-06bb-404b-8dcd-4dcb551bbee8

And, confirming the behaviour is correct upon reversion of the change:

https://github.com/google/site-kit-wp/assets/18395600/2bf9b31d-95f0-4ff5-8638-5a7838fef701

As a result, I'll go ahead and revert that commit, restoring the PR to how it was when I initially picked it up for MR. Considering that we've not seen any evidence of a flicker in our testing this seems like the way to go here, and we can just keep an eye out for it in future.

File: assets/js/components/DashboardNavigation/Navigation.js [hunk 3]
```diff
@@ -99,13 +99,21 @@ export default function Navigation() {
...
 	const widgetContextOptions = {
 		modules: viewableModules ? viewableModules : undefined,
 	};
 
 	const showKeyMetrics = useSelect( ( select ) => {
-		if ( ! userInputEnabled || dashboardType !== DASHBOARD_TYPE_MAIN ) {
+		if (
+			! userInputEnabled ||
+			dashboardType !== DASHBOARD_TYPE_MAIN ||
+			isKeyMetricsWidgetHidden === true
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/components/DashboardNavigation/Navigation.js:115]: As [discussed](https://github.com/google/site-kit-wp/pull/7247/files#r1263526791) it's evident we don't in fact want to make this change at this point...

**Review by techanvil — APPROVED** [review]
LGTM! Nice one @jimmymadon.
